src/screen/screen.cpp

```diff
@@ -0,0 +1,152 @@
+#include <iostream>
+#include "screen.h"
+
+// usings
+using std::cout;
+using std::endl;
+
+namespace vmonitor {
+// macros
+// delete and set null
+#define delete(x) { delete x; x = NULL; }
+#define delete_arr(x) { delete [] x; x = NULL; }
+
+	Screen::Screen():
+		m_window(NULL),
+		m_renderer(NULL),
+		m_texture(NULL),
+		m_buffer(NULL) {
+			// constructor body
+			if(!init())
+				close();
+		}
+
+	bool Screen::init() {
+		// SDL init
+		if(SDL_Init(SDL_INIT_VIDEO) < 0) {
+			print_err_msg("SDL init failed.");
+			return false;
+		}
+
+		// create window
+		m_window = 
+			SDL_CreateWindow("Particle Fire Explosion", 
+					SDL_WINDOWPOS_UNDEFINED, 
+					SDL_WINDOWPOS_UNDEFINED, 
+					SCREEN_WIDTH,
+					SCREEN_HEIGHT,
+					SDL_WINDOW_SHOWN);
+
+		// window null error
+		if(m_window == NULL) {
+			print_err_msg("SDL Window init failed.");
+			return false;
+		}
+
+		// syncing renderer
+		m_renderer = SDL_CreateRenderer(m_window, 
+				-1, 
+				SDL_RENDERER_PRESENTVSYNC);
+		m_texture = SDL_CreateTexture(m_renderer,
```

Comment:
You're using `m_renderer` without being sure it's actually allocated at this point. `STD_CreateRenderer` could fail?